scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -2997,6 +2997,17 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
       )
     val tparams = typeParamClauseOpt(ownerIsType = false, ctxBoundsAllowed = true)
     val paramss = paramClauses(ownerIsType = false).require[List[List[Term.Param]]]
+
+    def onlyLastParameterCanBeRepeated(params: List[Term.Param]): Unit = {
+      params.reverse.tail.foreach(p =>
```

Comment:
`params.init` is equivalent to `reverse.tail`. However, either approach allocates a lot of memory. We should avoid re-creating the list to do a check, we can do it like this
```scala
def checkRepeatedParams(params: List[Term.Param]): Unit =  params match {
  case Nil | _ :: Nil => // OK
  case param :: tail =>
    checkNotRepeated(param)
    checkRepeatedParams(params)
}
def checkRepeatedParamss(sparamss: List[List[Term.Param]]): Unit = paramss match {
  case Nil => // OK
  case params :: Nil => checkRepeatedParams(params)
  case params :: nonEmptyTail =>
    params.foreach(p => checkNonRepeated(p))
    onlyLastParameterCanBeRepeated(nonEmptyTail)
}
```

Reply:
Alternatively, 
```scala
val repeatedParam = paramss.iterator.map(_.iterator).flatten.find(isRepatedParam)
val lastParam = paramss.lastOption.flatMap(_.lastOption))
if (repeatedParam.nonEmpty && lastParam != lastOption) {
  boom()
}
```

Reply:
I changed `reverse.tail` because it throws exception for empty list (similar to `list.init`)

Reply:
@olafurpg to be clear what is expected here, we have `List[List[Param]]` and we want to ensure for each list only last Param in each list can be RepeatedType. There might be multiple RepeatedType parameters in the end e.g.:
`def fx(x1: Boolean, x2: Int*)(y1: String, y2: Int*)`
we will get: 
`List(List(Boolean, Int*), List(String, Int*))` - which is correct

Reply:
I misunderstood the expected behavior. I thought the following program was invalid but it's actually OK
```
object Foo {
  def foo(a: Int, b: Int*)(c: Int, d: Int*) = ???
}


```

Reply:
Not really a syntax you use everyday :)